public/app/plugins/panel/heatmap/fields.ts

```diff
@@ -65,6 +66,13 @@ export function prepareHeatmapData(
     return {};
   }
 
+  // cache field.state.displayName
+  frames.forEach((fr) => {
+    fr.fields.forEach((fl) => {
+      getFieldDisplayName(fl, fr, frames);
```

Comment:
did you add a utility for this recently?

Reply:
:facepalm: you don't actually expect me to remember what happened [a week ago](https://github.com/grafana/grafana/pull/71161), do you?